synapse/rest/media/v1/media_storage.py

```diff
@@ -178,29 +188,23 @@ async def ensure_media_is_in_local_cache(self, file_info: FileInfo) -> str:
             # Fetch is supposed to return an Awaitable, but guard against
             # improper implementations.
             if inspect.isawaitable(res):
-                res = await res
-            if res:
-                with res:
+                res = await res  # type: ignore
+            if res:  # type: ignore
+                with res:  # type: ignore
```

Comment:
This is unfortunate, but I don't know about solution here, other than perhaps marking `res` as `Any`?

Reply:
I had played it a bit to see if I could get it to understand what this is, but I'll try again. 👍

Reply:
I did a bit more experimenting and using `Any` worked! 👍 I updated the comments around this (and another location) to specify what the return type *should* be though! Thanks for the idea!